backend/src/server.ts

```diff
@@ -102,3 +102,4 @@ const createServer = (options?) => {
 }
 
 export default createServer
+export type Context = Awaited<ReturnType<ReturnType<typeof getContext>>>
```

Comment:
Why does `getContext` have to return two nested functions?

Reply:
Because it needs to have context, and requires to be a function to evaluate during runtime.
If you have a better way to inject dependencies, feel free to implement it